### PR TITLE
Use oshinko-core for cluster commands

### PR DESCRIPTION
**Pull request by tmckayus**
This change adds oshinko-core to the vendor directory and
uses it for all cluster operations.

### CODE REVIEW

File: pkg/cmd/cli/cmd/create.go [hunk 1]
```diff
@@ -54,8 +52,8 @@ func CmdCreate(f *clientcmd.Factory, reader io.Reader, out io.Writer) *cobra.Com
 		},
 	}
 
-	cmd.Flags().Int("masters", 1, "Numbers of workers in spark cluster")
-	cmd.Flags().Int("workers", 1, "Numbers of workers in spark cluster")
+	cmd.Flags().Int("masters", 0, "Numbers of workers in spark cluster")
+	cmd.Flags().Int("workers", 0, "Numbers of workers in spark cluster")
 	cmd.Flags().String("masterconfigdir", defaultsparkconfdir, "Config folder for spark master")
```
Comment (sub-mod) [inline pkg/cmd/cli/cmd/create.go]: should these folders locations flags be deleted ?

File: pkg/cmd/cli/cmd/cmd.go [hunk 2]
```diff
@@ -65,20 +70,8 @@ func (o *CmdOptions) Complete(f *osclientcmd.Factory, cmd *cobra.Command, args [
 	if cmd.Flags().Lookup("workerconfigdir") != nil {
 		o.WorkerConfigDir = kcmdutil.GetFlagString(cmd, "workerconfigdir")
 	}
-	if cmd.Flags().Lookup("masterconfig") != nil && kcmdutil.GetFlagString(cmd, "masterconfig") != "" {
```
Comment (sub-mod) [inline pkg/cmd/cli/cmd/cmd.go]: why is this deleted ?

File: pkg/cmd/cli/cmd/cmd.go [hunk 3]
```diff
@@ -65,20 +70,8 @@ func (o *CmdOptions) Complete(f *osclientcmd.Factory, cmd *cobra.Command, args [
 	if cmd.Flags().Lookup("workerconfigdir") != nil {
```
Comment (sub-mod) [inline pkg/cmd/cli/cmd/cmd.go]: should this be deleted?

File: pkg/cmd/cli/cmd/create.go [hunk 4]
```diff
@@ -67,160 +65,16 @@ func CmdCreate(f *clientcmd.Factory, reader io.Reader, out io.Writer) *cobra.Com
...
-		deleteCluster(o)
-		return ErrorString(wDepConfigMsg, err)
-	}
-
-	// If we've gotten this far, then likely the cluster naming is not in conflict so
-	// assume at this point that we should use a 500 error code
-	sc := o.KClient.Services(o.Project)
-	_, err = sc.Create(&mastersv.Service)
+	config := clusterconfigs.ClusterConfig{}
+	config.WorkerCount = o.WorkerCount
```
Comment (sub-mod) [inline pkg/cmd/cli/cmd/create.go]: not sure where WorkerCount value is assigned ? Assume you don't pass --worker then the value is 0.

**Comment by elmiko** [comment]
Subin's questions not withstanding, this patch works for me with the basic CRUD operations.

File: pkg/cmd/cli/cmd/create.go [hunk 5]
```diff
@@ -54,8 +52,8 @@ func CmdCreate(f *clientcmd.Factory, reader io.Reader, out io.Writer) *cobra.Com
 		},
 	}
 
-	cmd.Flags().Int("masters", 1, "Numbers of workers in spark cluster")
-	cmd.Flags().Int("workers", 1, "Numbers of workers in spark cluster")
+	cmd.Flags().Int("masters", 0, "Numbers of workers in spark cluster")
```
Comment (tmckayus) [inline pkg/cmd/cli/cmd/create.go]: This has to do with how oshinko-core handles cluster configurations.  The master and worker counts are part of a config object that looks like this:

    type ClusterConfig struct {
	    MasterCount int
	    WorkerCount int
	    Name string
	    SparkMasterConfig string
	    SparkWorkerConfig string
    }

In this object, Name is the name of a stored cluster config.
If a stored cluster config is named, that configuration is read first by oshinko-core. It may contain values for all of the other fields of a ClusterConfig, and all stored configs have defaults of WorkerCount = 1 and MasterCount = 1 if they are not specified (0 == not specified)

If no named config is specified, an internal default config is used with WorkerCount = 1.

Then, if any of the other fields are explicitly set in the config object passed to create, those values overwrite the values read from the stored config. Again, 0 == unspecified for counts, and "" == unspecified for string fields.

So if we make the default value for --workers and --masters be 0, the clusters will pick up the defaults from oshinko or from the stored config.  If we leave them set to 1 explicitly, they will overwrite the default settings or a value from a stored config.

File: pkg/cmd/cli/cmd/create.go (hunk 1 above)
Comment (tmckayus) [inline pkg/cmd/cli/cmd/create.go]: Yes, I'll take those out.  SPARK_CONF_DIR should be left alone and set by oshinko-core to a fixed path iff a master or worker config is named.

File: pkg/cmd/cli/cmd/create.go (hunk 4 above)
Comment (tmckayus) [inline pkg/cmd/cli/cmd/create.go]: Correct, see above.  If worker is 0 then oshinko-core will default it appropriately.

File: pkg/cmd/cli/cmd/cmd.go (hunk 3 above)
Comment (tmckayus) [inline pkg/cmd/cli/cmd/cmd.go]: yes

File: pkg/cmd/cli/cmd/cmd.go (hunk 2 above)
Comment (tmckayus) [inline pkg/cmd/cli/cmd/cmd.go]: oshinko-core already scans for the presence of a named master config or worker config and generates an error, so we should let oshinko-core do that instead of checking early in the CLI code. The error will be generated before any objects are created.

File: pkg/cmd/cli/cmd/create.go (hunk 4 above)
Comment (tmckayus) [inline pkg/cmd/cli/cmd/create.go]: There is a wrinkle, however. Have to think about the scale case -- default results in an error, since it tries to scale to zero rather than 1. There might be an issue in oshinko-core, though, so I would call this a non-issue for the moment.  One could argue that scale without a count dropping to "1" is not something you want anyway.  So maybe the arg shold be required.

Don't see this is an issue to delay merge, though.
